FireRender.Maya.Src/FireRenderProduction.cpp

```diff
@@ -249,163 +255,180 @@ bool FireRenderProduction::start()
 
-		// Allocate memory for active AOV pixels and get
-		// the AOV that will be displayed in the render view.
-		m_aovs->setRegion(region, contextWidth, contextHeight);
-		m_aovs->allocatePixels();
-		m_renderViewAOV = &m_aovs->getRenderViewAOV();
+	return true;
+}
 
-		if (showWarningDialog)
-		{
-			rcWarningDialog.show();
-		}
+bool FireRenderProduction::startTileRender()
+{
+	assert(m_globals.tileRenderingEnabled);
 
-		// Start the Maya render view render.
-		startMayaRender();
+	int contextWidth = m_globals.tileSizeX;
+	int contextHeight = m_globals.tileSizeY;
 
-		// Ensure the scheduled update flag is clear initially.
-		m_renderViewUpdateScheduled = false;
+	RenderRegion region = RenderRegion(contextWidth, contextHeight);
 
-		m_isRunning = true;
+	Init(contextWidth, contextHeight, region);
+	FireRenderThread::KeepRunning([this]()
+	{
+		try
+		{
+			RenderTiles();
+			stop();
+			m_isRunning = false;
+		}
+		catch (...)
+		{
+			// We should stop rendering if some exception(error on core side) occured. It will close all progress bars etc
+			stop();
+			m_isRunning = false;
+			m_error.set(current_exception());
+		}
+		return m_isRunning;
+	});
 
-		SetupWorkProgressCallback();
+	return true;
+}
+// -----------------------------------------------------------------------------
+bool FireRenderProduction::startFullFrameRender()
+{
+	assert(!m_globals.tileRenderingEnabled);
 
-		refreshContext();
+	int contextWidth = m_width;
+	int contextHeight = m_height;
 
-		m_needsContextRefresh = false;
+	RenderRegion region = m_region;
 
-		m_progressBars->SetRenderingText(true);
-		m_contextPtr->setStartedRendering();
+	Init(contextWidth, contextHeight, region);
 
-		// Start the render
-		FireRenderThread::KeepRunning([this]()
+	// Start the render
+	FireRenderThread::KeepRunning([this]()
+	{
+		try
 		{
-			try
-			{
-				if (m_globals.tileRenderingEnabled)
-				{
-					RenderTiles();
-					stop();
-					m_isRunning = false;
-				}
-				else
-				{
-					m_isRunning = RunOnViewportThread();
-				}
-			}
-			catch (...)
-			{
-				// We should stop rendering if some exception(error on core side) occured. It will close all progress bars etc
-				stop();
-				m_isRunning = false;
-				m_error.set(current_exception());
-			}
+			m_isRunning = RunOnViewportThread();
+		}
+		catch (...)
+		{
+			// We should stop rendering if some exception(error on core side) occured. It will close all progress bars etc
+			stop();
+			m_isRunning = false;
+			m_error.set(current_exception());
+		}
 
-			return m_isRunning;
-		});
+		return m_isRunning;
+	});
 
-		m_NorthStarRenderingHelper.Start();
-	}
+	m_NorthStarRenderingHelper.Start();
 
-	return ret;
+	return true;
 }
 
-void FireRenderProduction::OnBufferAvailableCallback()
+void FireRenderProduction::OnBufferAvailableCallback(float progress)
 {
 	AutoMutexLock pixelsLock(m_pixelsLock);
-	m_renderViewAOV->readFrameBuffer(*m_contextPtr, true);
+
+	bool frameFinished = fabs(1.0f - progress) <= FLT_EPSILON;
```

Comment:
hmm not sure about such check

Reply:
richardgeslot  7:57 PM
Hi Dmitrii, yes 0.0 to 1.0
7:58
even for multiple iterations :
rprContextSetParameterByKey1u(context_, RPR_CONTEXT_ITERATIONS, N);
this callback will be 0.0 to 1.0